src/timebar/charts/activity.js

```diff
@@ -1,56 +1,125 @@
-import React, { Component } from 'react'
+import React, { useContext, useMemo } from 'react'
 import PropTypes from 'prop-types'
-import { area, curveStepAfter } from 'd3-shape'
-import { animated, Spring } from 'react-spring/renderprops'
+import maxBy from 'lodash/maxBy'
+import max from 'lodash/max'
+import {
+  area,
+  curveStepAfter,
+  curveStepBefore,
+  curveLinear,
+  curveBasis,
+  curveCatmullRom,
+  curveCardinal,
+} from 'd3-shape'
 import { getTime } from '../utils'
 import ImmediateContext from '../immediateContext'
-import styles from './activity.module.css'
+import { DEFAULT_CSS_TRANSITION } from '../constants'
 
 const TOP_MARGIN = 5
 const BOTTOM_MARGIN = 20
 
-class Activity extends Component {
-  static contextType = ImmediateContext
-  render() {
-    const { activity, absoluteEnd, outerScale, outerWidth, graphHeight } = this.props
-    const { immediate } = this.context
-
-    // because data stops at last day midnight, add an extra day with the same data
-    // to allow the curve to go to the end of the day
-    const lastDay = activity[activity.length - 1]
-    const addedLastDay = { date: getTime(absoluteEnd), value: lastDay.value }
-
-    const finalHeight = graphHeight - TOP_MARGIN - BOTTOM_MARGIN
-    const middle = TOP_MARGIN + finalHeight / 2
-
-    const areaGenerator = area()
-      .x((d) => outerScale(d.date))
-      .y0((d) => middle - (finalHeight * d.value) / 2)
-      .y1((d) => middle + (finalHeight * d.value) / 2)
-      .curve(curveStepAfter)
-
-    return (
-      <svg width={outerWidth} height={graphHeight} className={styles.Activity}>
-        <Spring native immediate={immediate} to={{ d: areaGenerator([...activity, addedLastDay]) }}>
-          {(style) => <animated.path d={style.d} fill="pink" fillOpacity={0.9} />}
-        </Spring>
-      </svg>
-    )
-  }
+const CURVES = {
+  curveStepAfter,
+  curveStepBefore,
+  curveLinear,
+  curveBasis, // smoothes out, line does not pass through points
+  curveCatmullRom, // smoothes out while insuring line passes through points but produces artefacts
+  curveCardinal, // smoothes out while insuring line passes through points but produces artefacts
+}
+
+const getMaxValue = (activity) => {
+  const maxValues = activity.map((segment) => {
+    return maxBy(segment, (item) => item.value).value
+  })
+  const maxValue = max(maxValues)
+  return maxValue
+}
+
+const getPaths = (graphHeight, activity, absoluteEnd, overallScale, maxValue, curve) => {
+  const finalHeight = graphHeight - TOP_MARGIN - BOTTOM_MARGIN
+  const middle = TOP_MARGIN + finalHeight / 2
+
+  const areaGenerator = area()
+    .x((d) => overallScale(d.date))
+    .y0((d) => middle - (finalHeight * d.value) / maxValue / 2)
+    .y1((d) => middle + (finalHeight * d.value) / maxValue / 2)
+    .curve(CURVES[curve])
+
+  const paths = activity.map((segment, i) => {
+    // if (i === activity.length - 1) {
```

Comment:
@nerik as I was mentioning above, try to avoid commented code or explain what it should be used for